City-of-Helsinki/helsinki-paatokset
# PP-233: Policymakers integration with Ahjo API

NOTE: The data aggregation from Ahjo API production doesn't currently work because of connectivity issues. That will be tested later. Cleanup and duplicate content handling (and porting over old manually entered content for policymakers) will be handled in a separate ticket.

Branch is based on this PR: https://github.com/City-of-Helsinki/helsinki-paatokset/pull/82, which should be merged first.

**To test**
* Run `make new` (or `drush cr; drush cim -y`)
* Run `drush ap:fs -v`
* Run ´drush migrate-status --tag=ahjo_api_decisionmakers´
  * This should show one available migration with 696 non imported entities
* Run `drush migrate-import ahjo_decisionmakers:all`
* Check some of the migrated content: https://helsinki-paatokset.docker.so/fi/admin/content?title=&type=policymaker&status=All&langcode=All
  * Note: If you have imported data from `paatokset_policymakers`, then there will be duplicates
  * The theme doesn't fully work yet (org colors, etc), but this will be handled in the cleanup ticket
  * There are some office holders with duplicate titles (but different IDs). Fixes related to these entities will be handled in the cleanup ticket.
* Read code changes, is there anything that should be changed or doesn't make sense?
